Src/DependencyCollector/Shared/Implementation/ClientServerDependencyTracker.cs

```diff
@@ -32,6 +32,18 @@ internal static DependencyTelemetry BeginTracking(TelemetryClient telemetryClien
             // but every operation must have it's own Activity and Id must be set accordingly
             // basically we repeat TelemetryClientExtensions.StartOperation here
             var activity = new Activity(DependencyActivityName);
+
+            // This is workaround for the issue https://github.com/Microsoft/ApplicationInsights-dotnet/issues/538
+            // if there is no parent Activity, ID Activity generates is not random enough to work well with 
+            // ApplicationInsights sampling algorithm
+            // This code should go away when Activity is fixed: https://github.com/dotnet/corefx/issues/18418
+            if (Activity.Current == null)
```

Comment:
Can we move this logic into `Start` method in Base Application Insights SDK so everybody using `Start` & `Stop` will have sampling working as expected?